tap_kafka/__init__.py

```diff
@@ -40,20 +42,32 @@ def dump_catalog(all_streams):
 def do_discovery(config):
     """Discover kafka topic by trying to connect to the topic and generate singer schema
     according to the config"""
-    consumer = KafkaConsumer(config['topic'],
-                             group_id=config['group_id'],
-                             enable_auto_commit=False,
-                             consumer_timeout_ms=config.get('consumer_timeout_ms', DEFAULT_CONSUMER_TIMEOUT_MS),
-                             bootstrap_servers=config['bootstrap_servers'].split(','))
+    consumer = Consumer({
+        'bootstrap.servers': config['bootstrap_servers'],
+        'group.id': config['group_id'],
+        'auto.offset.reset': 'earliest',
+    })
+
+    try:
+        topic = config['topic']
+
+        LOGGER.info(f"Discovering {topic} topic...")
+        cluster_md = consumer.list_topics(topic=topic, timeout=config['session_timeout_ms'] / 1000)
```

Comment:
this is making `session_timeout_ms` required when it's not according to the README.
There is a fallback default value to use `DEFAULT_SESSION_TIMEOUT_MS`

Reply:
The `config` dict here is not the original `config.json` but a modified one where all the default values are replaced. This parameter should always exist at this point.

The original config modified by the [generate_config](https://github.com/transferwise/pipelinewise-tap-kafka/blob/2501966faf7a2afd8ecbeb8b3a875cd0a7ac7498/tap_kafka/__init__.py#L77) function and the output is passed to the [do_discover](https://github.com/transferwise/pipelinewise-tap-kafka/blob/2501966faf7a2afd8ecbeb8b3a875cd0a7ac7498/tap_kafka/__init__.py#L108) function

Reply:
aha! my bad!